src/lib.rs

```diff
@@ -146,7 +371,11 @@ pub fn get_bundle_identifier(app_name: &str) -> Option<String> {
 /// Set the application which delivers or schedules a notification
 pub fn set_application(bundle_ident: &str) -> NotificationResult<()> {
     unsafe {
-        ensure!(!APPLICATION_SET, ApplicationError::AlreadySet(bundle_ident.into()));
+        // TODO: Is there a point to restricting set_application to only once?
```

Comment:
Yes there is. At least at the time I wrote this. It has something to do with "faking" the bundle identifier. Apparently the NotificationDelegate not always work if you change the bundle_identifier once you scheduled a notification.